### PR TITLE
Fix sequentially correlated set generator.

### PR DESCRIPTION
**Pull request by huangxichen1**
1. Sequentially correlated set generator will run into error if the new set is smaller than the overlap set size. This PR fix this bug by capping the size of the overlap set if it is larger than the new set. Also add a unit test for this feature.
2. The sequentially correlated set generator does not need the universe size. Remove this argument in order to reduce potential bug when creating the evaluation configuration. Also modified the interoperability test and the evaluation configs and its tests.

### CODE REVIEW

File: src/simulations/set_generator.py [hunk 1]
```diff
@@ -331,29 +332,33 @@ def __iter__(self):
...
+
 class _SequentiallyCorrelatedAllPreviousSetGenerator(SetGeneratorBase):
   """Generator for sequentailly correlated sets.
 
   Each set has some overlap with the union of previously generated campaigns.
   """
 
-  def __init__(self, universe_size, shared_prop, set_size_list, random_state):
-    self.universe_size = universe_size
+  def __init__(self, shared_prop, set_size_list, random_state):
```
Comment (kungfucraig) [inline src/simulations/set_generator.py:342]: Can you document the input params?

File: src/simulations/set_generator.py [hunk 2]
```diff
@@ -331,29 +332,33 @@ def __iter__(self):
...
-                                 for set_size in self.set_size_list[:-1]]
-                              + [0])
-    # self.overlap_size_list[i] is the overlap of set.set_size_list[i]
-    # with the previous union, so in particular, self.overlap_size_list[0] = 0
-    total_ids_size = int(sum(self.set_size_list) - sum(self.overlap_size_list))
-    self.ids_pool = choice_fast(universe_size, total_ids_size, self.random_state)
+    self.num_sets = len(set_size_list)
+    self.overlap_size_list = [0]
+    total_ids_size = set_size_list[0]
+    for i in range(self.num_sets - 1):
```
Comment (kungfucraig) [inline src/simulations/set_generator.py:349]: Can you add comments about what you're doing here. Basically just transfer the description from the CL. I think it's not straightforward why this code exists.

File: src/simulations/tests/set_generator_test.py [hunk 3]
```diff
@@ -407,13 +412,30 @@ def test_sequentially_correlated_all_previous_generator_raise_error(self):
...
 
+  @parameterized.parameters(
+      (set_generator.CORRELATED_SETS_ALL,),
+      (set_generator.CORRELATED_SETS_ONE,))
+  def test_sequentially_correlated_generator_overlap_size_larger_than_set_size(
+      self, correlation_type):
+    sc_gen = set_generator.SequentiallyCorrelatedSetGenerator(
+        order=set_generator.ORDER_ORIGINAL, correlated_sets=correlation_type,
+        shared_prop=0.5, set_sizes=[10, 1],
+        random_state=np.random.RandomState())
```
Comment (kungfucraig) [inline src/simulations/tests/set_generator_test.py:430]: You sure you don't want a constant seed here?

File: src/simulations/set_generator.py (hunk 1 above)
Comment (huangxichen1) [reply src/simulations/set_generator.py:342]: Done.

File: src/simulations/tests/set_generator_test.py (hunk 3 above)
Comment (huangxichen1) [reply src/simulations/tests/set_generator_test.py]: Added a constant seed.

File: src/simulations/set_generator.py (hunk 2 above)
Comment (huangxichen1) [reply src/simulations/set_generator.py:363]: Done.

**Comment by jiayu-google** [comment]
Thank Xichen for further making the codes consistent!

Now the shared proportion means overlap / **current_set_size**, for both the "correlated with all" and "correlated with THE previous" sub scenarios.

**Pull request merged**